precompile/allowlist/allowlist.go

```diff
@@ -119,11 +123,20 @@ func createAllowListRoleSetter(precompileAddr common.Address, role Role) contrac
 
 		// Verify that the caller is an admin with permission to modify the allow list
 		callerStatus := GetAllowListStatus(stateDB, precompileAddr, callerAddr)
-		if !callerStatus.IsAdmin() {
+		// At this point we know that Manager role is activated.
+		if IsManagerRoleActivated(evm) && callerStatus == ManagerRole {
+			// Get current status.
+			// Before the manager role, we never checked the status of the address we are trying to modify.
+			// So we should keep the same behaviour by special casing this.
+			modifyStatus := GetAllowListStatus(stateDB, precompileAddr, modifyAddress)
+			if !callerStatus.CanModify(IsManagerRoleActivated(evm), modifyStatus, role) {
```

Comment:
nit: I think if we are only calling `CanModify` here inside the check for the manager role (to avoid reading the stateDB for the previous role, I assume), should we call it `CanManagerModify` and remove the other clauses from that function? I think otherwise it may give the impression that `CanModify` controls the behavior for other roles as well.
Also it is not needed to call `IsManagerRoleActivated` again. Possibly consider inlining the function.

Reply:
I think it would be dangerous to provide the `CanModify` function without taking the `isManagerActivated` as we export those. if someone uses it before the activation that could potentially go wrong.

Reply:
nit: seems we could assign mgrRoleActivated to a variable to avoid calling it twice or simply pass true then?

Reply:
added `isManagerRoleActivated` variable on top of the function and used it here.